org.adempiere.base/src/org/adempiere/base/sso/ISSOPrinciple.java

```diff
@@ -0,0 +1,49 @@
+/******************************************************************************
+ * Copyright (C) 2016 Logilite Technologies LLP								  *
+ * This program is free software; you can redistribute it and/or modify it    *
+ * under the terms version 2 of the GNU General Public License as published   *
+ * by the Free Software Foundation. This program is distributed in the hope   *
+ * that it will be useful, but WITHOUT ANY WARRANTY; without even the implied *
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.           *
+ * See the GNU General Public License for more details.                       *
+ * You should have received a copy of the GNU General Public License along    *
+ * with this program; if not, write to the Free Software Foundation, Inc.,    *
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.                     *
+ *****************************************************************************/
+package org.adempiere.base.sso;
+
+import java.io.IOException;
+import java.text.ParseException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.compiere.util.Language;
+
+/**
+ * @author Logilite Technologies
+ *
+ */
+public interface ISSOPrinciple
```

Comment:
This represent a Principle Service instead of a Principle entity and should be name accordingly. In other words, the current naming is confusing and should be named as "ISSOPrincipleService" instead. The other common name use is "Identity Provider".

Reply:
Renamed to ISSOPrincipalService